transactor/src/test/scala/io/mediachain/copycat/SeedingCopycatClient.scala

```diff
@@ -0,0 +1,97 @@
+package io.mediachain.copycat
+
+import cats.data.Xor
+import io.mediachain.copycat.Client.{ClientState, ClientStateListener}
+import io.mediachain.copycat.Dummies.DummyReference
+import io.mediachain.protocol.Datastore.{CanonicalRecord, ChainCell, ChainEntry, _}
+import io.mediachain.protocol.Transactor.{JournalCommitError, JournalError, JournalListener}
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+
+class SeedingCopycatClient(serverAddress: String)
+  (implicit executionContext: ExecutionContext)
+  extends JournalListener {
+
+  val client = Client.build()
+
+  def replayEntry(journalEntry: JournalEntry, client: Client, datastore: Datastore)
+    (implicit executionContext: ExecutionContext)
+  : Future[Xor[JournalError, JournalEntry]] = {
+    journalEntry match {
+      case e: CanonicalEntry => {
+        val canonical =
+          datastore.getAs[CanonicalRecord](e.ref)
+
+        canonical.map(c => client.insert(c)
+            .map { res =>
+              println(s"replayed entry ${journalEntry.index}. result: $res")
+              res
+            }
+        )
+          .getOrElse(
+            Future.failed(
+              new RuntimeException("Unable to replay journal entry: " +
```

Comment:
We should be careful about how we mix futures and Xors. I think we can either put the errors in the Xor.left, or in the Future.failed, just as long as we're consistent with it.